src/lib.rs

```diff
@@ -168,68 +218,19 @@ pub mod ion_hash;
 pub mod lazy;
 // Experimental Streaming APIs
 mod position;
+mod read_config;
 #[cfg(feature = "experimental-serde")]
 pub mod serde;
 pub(crate) mod unsafe_helpers;
-
-pub use catalog::{Catalog, MapCatalog};
-pub use element::builders::{SequenceBuilder, StructBuilder};
-pub use element::{
-    reader::ElementReader, writer::ElementWriter, Annotations, Element, IntoAnnotatedElement,
-    IntoAnnotations, Sequence, Value,
-};
-pub use ion_data::IonData;
-pub use shared_symbol_table::SharedSymbolTable;
-pub use symbol_ref::SymbolRef;
-#[doc(inline)]
-pub use types::{
-    decimal::Decimal, Blob, Bytes, Clob, Int, IonType, List, Null, SExp, Str, Struct, Symbol,
-    SymbolId, Timestamp, TimestampPrecision, UInt,
-};
-
-// Allow access to less commonly used types like decimal::coefficient::{Coefficient, Sign}
-pub use types::decimal;
-
-// These re-exports are only visible if the "experimental-reader" feature is enabled.
-#[cfg(feature = "experimental-reader")]
-pub use {
-    binary::non_blocking::raw_binary_reader::RawBinaryReader,
-    blocking_reader::{BlockingRawBinaryReader, BlockingRawReader, BlockingRawTextReader},
-    ion_reader::IonReader,
-    raw_reader::{BufferedRawReader, RawReader, RawStreamItem},
-    raw_symbol_token::RawSymbolToken,
-    raw_symbol_token_ref::RawSymbolTokenRef,
-    // Public as a workaround for: https://github.com/amazon-ion/ion-rust/issues/484
-    reader::integration_testing,
-    reader::{Reader, ReaderBuilder, StreamItem, UserReader},
-    symbol_table::SymbolTable,
-    system_reader::{SystemReader, SystemStreamItem},
-    text::non_blocking::raw_text_reader::RawTextReader,
-};
-
-// These re-exports are only visible if the "experimental-writer" feature is enabled.
-#[cfg(feature = "experimental-writer")]
-pub use lazy::encoder::writer::ApplicationWriter;
-
-// Exposed to allow benchmark comparisons between the 1.0 primitives and 1.1 primitives
-#[cfg(feature = "experimental-lazy-reader")]
-pub use {
-    binary::int::DecodedInt, binary::non_blocking::type_descriptor::Header,
-    binary::uint::DecodedUInt, binary::var_int::VarInt, binary::var_uint::VarUInt,
-    element::writer::WriteConfig, lazy::binary::immutable_buffer::ImmutableBuffer,
-    lazy::encoder::binary::v1_1::flex_int::FlexInt,
-    lazy::encoder::binary::v1_1::flex_uint::FlexUInt,
-};
-
-#[doc(inline)]
-pub use result::{IonError, IonResult};
+mod write_config;
 
 /// Whether or not the text spacing is generous/human-friendly or something more compact.
-#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+#[derive(Copy, Clone, Debug, Eq, PartialEq, Default)]
 #[non_exhaustive]
 pub enum TextKind {
     Compact,
     Lines,
+    #[default]
```

Comment:
🗺️ In my own tests I found that I wanted `Pretty` most of the time. Switching to more compact forms is easy.